cluster.c

```diff
@@ -678,15 +674,6 @@ void ShardingInfoRDBLoad(RedisModuleIO *rdb)
      * be zero).
      */
     unsigned int rdb_shard_groups_num = RedisModule_LoadUnsigned(rdb);
-    if (!rdb_shard_groups_num) {
-        /* No shardgroups. This could mean no sharding, or simply no shardgroups
-         * to read. If we have ShardingInfo we'll reset it.
-         */
-        if (si)
-            ShardingInfoReset(rr);
-
-        return;
-    }
```

Comment:
this is not neccessary anymore as a) rdb_shard_group_num will always be at least 1 and the loop below will handle local fine (and would still handle it fine even if it was 0).  b) si can no longer be null (older change) and hence the loop below would also work fine and dont have to special case it.